### ISSUE TITLE
Broken visualization

### ISSUE THREAD

**Issue by vladimir-popov** [vladimir-popov/forgotten_catacomb#1]
Seed: 6199650155624857372
```
╔═════════════════════════════════════╦═══════════╗
║                                     ║           ║
║                                  ║ HP: 100   ║
║     #################            ║           ║
║######......r........#            ║           ║
║...r.'...............#            ║           ║
║######...............#            ║           ║
║#    #...............#            ║           ║
║#    #............@..#            ║           ║
║#    #################            ║           ║
║                                  ║           ║
║                                  ║           ║
║                                  ║           ║
║                                  ║           ║
║                                  ║           ║
╚═════════════════════════════════════╩═══════════╝
```

**Comment by vladimir-popov** [comment]
More cases (seed 15776927952299183702):
```
╔═══════════════════════════════╦════════╗
║                               ║        ║
║      ####################     ║HP: 100 ║
║#######..................#     ║        ║
║......+.......r7mr..........#     ║        ║
║#######..................#     ║Rat     ║
║      #r...........r.....#     ║10      ║
║      #..................#     ║        ║
║      #...r..............#     ║        ║
║      ####################     ║        ║
║                               ║        ║
║                               ║Attack  ║
╚═══════════════════════════════╩════════╝


╔═══════════════════════════════╦════════╗
║                               ║        ║
║      ####################     ║HP: 100 ║
║#######..................#     ║        ║
║......+.......r..........#     ║        ║
║#######..................#     ║Rat     ║
║      #r..........@r.....10    ║        ║
║      #..................#     ║        ║
║      #...r..............#     ║        ║
║      ####################     ║        ║
║                               ║        ║
║                               ║Attack  ║
╚═══════════════════════════════╩════════╝
```

**Comment by vladimir-popov** [comment]
fixed in ce39e5a51c14673a2a89bcfdf70637a8f79e519f
